internal/kafka/internal/workers/kafka_mgrs/promotion/chain/reconciler_chain.go

```diff
@@ -0,0 +1,69 @@
+package chain
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+)
+
+// ReconcileAction - This is the interface that must be implemented by each action that the reconciler will execute.
+// T - is the type of result the action will generate
+type ReconcileAction[T any] interface {
+	// PerformJob - This is the method where the Action implementation will perform the real job
+	// * kafkaRequest - the kafka request to reconcile
+	// * currentResult - object containing the results that have been produced up to this point into the pipeline
+	// Return values
+	// * res Result[T] - the result produced by this action
+	// * finished bool - returning `true` here will interrupt the pipeline execution
+	// * err error - an eventual error. Returning an error always interrupts the pipeline execution
+	PerformJob(kafkaRequest *dbapi.KafkaRequest, currentResult Result[T]) (res Result[T], finished bool, err error)
+}
+
+// Result - This struct contains the result produced by each action
```

Comment:
To be idiomatic with Go documentation style this should be reworded as:
`Result contains ...`